### PR TITLE
while loop cleanup and wait_for function improvements

### PR DESCRIPTION
**Pull request by jdsn**
cleaning up all while loops and endless loops in qa_crowbarsetup to prevent hanging mkcloud runs due to loops that never finish

### CODE REVIEW

**Comment by bmwiedemann** [comment]
:+1:

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -1574,14 +1571,8 @@ function onadmin_allocate()
...
-        sleep 10
-    done
-    echo "Found one node"
-    while test $(get_all_discovered_nodes | wc -l) -lt $nodenumber ; do
-        sleep 10
-    done
+    wait_for 50 10 'test $(get_all_discovered_nodes | wc -l) -ge 1' "first node to be discovered"
+    wait_for 100 10 '[[ $(get_all_discovered_nodes | wc -l) -ge $nodenumber ]]' "all nodes to be discovered"
     local n
     for n in `get_all_discovered_nodes` ; do
```
Comment (toabctl) [inline scripts/qa_crowbarsetup.sh]: Hm. Aren't we waiting here twice for all nodes to be discovered? line 1575 is also waiting. Or am I missing something?

Comment (toabctl) [inline scripts/qa_crowbarsetup.sh]: Ah. ```get_all_discovered_nodes``` doesn't list all nodes in ```discovered``` state (which I assumed). Seems to be just confusing wording...

**Comment by toabctl** [comment]
+1

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: First wait is just for the first node, the second is for all nodes. This helped in the past a lot to see if at least one node was discovered or none.

Concerning the wording: discovered meaning `crowbar machines list | grep "^d"` lists it. So the node was seen by crowbar and no longer has an "h" as prefix.

**Pull request merged**